include/merlin/external_storage.cuh

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <cstdint>
+#include <type_traits>
+#include "merlin/memory_pool.cuh"
+
+namespace nv {
+namespace merlin {
+
+template <class Key, class Value>
+class ExternalStorage {
```

Comment:
Expect the local storage, it's expected that the set of APIs can be compatible with some remote parameter servers on end-user side. So there should be better name for the class?

Reply:
Well, I am open for suggestions.

* `ExternalStorage` (current)
* `SecondaryStorage`
* `OffloadingStorage`
* `AuxilliaryStorage`

I can live with either of these.

Reply:
Make a suggesion ;-)!

Reply:
Scope of PR changed.